src/types.jl

```diff
@@ -481,6 +550,72 @@ function create_entry(parent::LHDataStore, name::AbstractString,
     nothing
 end
 
+# write NTuple
+function create_entry(parent::LHDataStore, name::AbstractString, 
+    data::T; kwargs...) where {L, U, T <: NTuple{L, U}}
+    
+    create_entry(parent, name, reinterpret(U, [data]); kwargs...)
+    setdatatype!(parent.data_store[name], T)
+    nothing
+end
+
+# write Array{<:NTuple}
+function create_entry(parent::LHDataStore, name::AbstractString, 
+    data::AbstractArray{T, N}; kwargs...) where {N, L, U, T <: NTuple{L, U}}
+
+    create_entry(parent, name, _flatview_of_array_of_ntuple(data); kwargs...)
+    setdatatype!(parent.data_store[name], Array{NTuple{L, U}, N})
+    nothing
+end
+
+# write EncodedArray
+function create_entry(parent::LHDataStore, name::AbstractString, 
+    data::T; kwargs...) where {C, U, T <: EncodedArray{U, 1, C}}
+
+    create_entry(parent, name*"/encoded_data", data.encoded; kwargs...)
+    create_entry(parent, name*"/size", data.size)
+
+    # quick fix for avoiding hardcoding eltype while reading EncodedArray's
+    parent[name*"/sample_data"] = U(1.0) 
```

Comment:
Can you quickly elaborate on this?

Reply:
I was referring to https://github.com/legend-exp/LegendHDF5IO.jl/blob/1e64ddd03c7bbb3db2693a1f3231233b2ffbeb53/src/generic_io.jl#L555